### ISSUE TITLE
Feature Request: Highlight images as notes on OSM 

### ISSUE THREAD

**Issue by DaCor1** [mapillary/mapillary_issues#629]
A feature I would love to see is where I can select an image in my gallery and have an option to tick a box etc to force the image to appear as a link within a note in OSM notes layer. It would also be good to be able to add a note along with the link to the image on Mapillary

See here as an example done with @OSMThis

http://www.openstreetmap.org/note/180352#map=17/53.32862/-8.18515&layers=N

**Comment by gyllen** [comment]
I think a better solution would be to integrate Mapillary as a whole in to JOSM. What do you think about that?

**Comment by DaCor1** [comment]
Well yes that would be great but that is a completely separate issue

**Comment by gyllen** [comment]
So you would auth against openstreetmap and then automatically push from Mapillary website?

**Comment by DaCor1** [comment]
Yes, all you are pushing from Mapillary is text. The text in the note contains a link back to the image on the Mapillary site along with anything the user added as additional information as per my link above in the @osmthis example

**Comment by de-vries** [comment]
@DaCor1 How do you see the usage of such a feature? cause in every single Mapillary photo there is something that can be mapped in osm. You might as well ask osm to implement a Mapillary layer.

**Comment by de-vries** [comment]
@gyllen With `http://www.openstreetmap.org/note/new#map=<zoom level>/<long>/<lat>` you can make a link to a new osm-note.

for example: http://www.openstreetmap.org/note/new#map=51.59304/4.88387

**Comment by gyllen** [comment]
Im not too familiar with JOSM. But for me it seems smarter to implement a Mappilary layer in JOSM and then have notes link to Mapillary images instead (if needed). Is it possible to include a Mapillary link in a OSM note?

**Comment by de-vries** [comment]
> But for me it seems smarter to implement a Mappilary layer in JOSM

would lave to see that happen
> Is it possible to include a Mapillary link in a OSM note

Sorry can' t give you the answer. The link I gave you is about it for my coding skills :D

**Comment by DaCor1** [comment]
It is possible to include a mapillary link in a note

Just to clarify the idea behind this, think of it along the line of the traffic signs feature you have recently implemented where the user is drawn to a specific point on the map because there is something of note located there. 

What i am suggesting would work in a similar manner. 

When the app is running, and you spot something significant, in order to highlight it to mappers to a greater degree, have another icon on the screen (e.g. a pen & paper, or a star etc), which, when pressed, will automagically flag the next image taken as one that should be linked to in a note within OSM. 

As there are many more people monitoring notes than there are users of Mapillary, this would likely drive users to use the service to an even greater degree

**Comment by vincentdephily** [comment]
I'd like to see that too, I mentioned it in an email on osm talk yesterday and am just stumbling upon this github issue today.

The usecase I see is OSM notes on steroids thanks to mapillary. Say I notice a map error (wrong street name) during a stroll. I could use OSMand/vespucci/etc to create a note with the street's proper name, or I could use mapillary to take a photo of the street sign and just write "fix street name" as the note text. To be clear, this is about opportunistic note/mapillary usage, not (necessarily) when I was doing a mapillary sequence or an osm survey to begin with.

Creating notes like this is faster than typing the street name yourself (especially here in Ireland with bilingual street signs and diacritics), is more verifyable by the person who'll close the note, and contributes one more mapillary photo.

**Comment by gyllen** [comment]
I like this idea. Would this be possible to implent as a standalone feature using only the APIs from mapillary and OSM @vincentdephily @de-vries @DaCor1

**Comment by DaCor1** [comment]
I don't have the skills but it should be possible to have an option inside
Mapillary where there could be an additional icon on the screen,which, when
pressed would create a note through the OSM notes api with a link to the
next image taken in the app.

The note creation would have to happen on the Mapillary server side once
the images are uploaded.

Maybe something as simple as clicking that button would write some kind of
flag to the exif data which would be picked up server side and would then
automatically generate an OSM note.

The content of the note only needs to be a link to the image on the
Mapillary site. There might be scope for adding text if in single shot
mode, but initially I think a simple link would suffice if adding a text
option is too messy

DaCor

**Comment by gyllen** [comment]
@DaCor1 Ah I finally get the idea now. Basically you want to

1. Create a note on OSM and ask for a mapillary image.
2. When an image according to that note is uploaded it would automatically answer the other OSM note that requested the image.

Do I understand things correctly?

**Comment by DaCor1** [comment]
Umm nope lol.

I don't know how to explain it better than I have.

Essentially the flow goes

1. Click a button in the app
2. Next image taken is the one that will be linked to from the OSM note and
is denoted as being "special" in some way (add something to exif data
maybe?)
3. Upload images to server
4. Script on server detects this image as "special"
5. Script creates a note on OSM with the content of the note being just the
url to the "special" image in Mapillary

Does that clear it up?

Sorry I'm trying my best to explain.

**Comment by gyllen** [comment]
Get it now :). Its a very specific feature would need to see how many that wants this. Its pretty cool though. People comment below if you want this feature.

**Comment by kaitu** [comment]
Cool. I'm otherwise afraid that an image alone might not be explicative enough. OSM is already getting jammed with notes which are unresolvable due to an excessively cryptic message, I reckon the app should ask for a mandatory explanatory comment. A connection to OSM user credentials might also be good, so that the note will not be anonymous. 

I like the idea, I'd rather use a desktop tool rather than an app button, that would also work for images uploaded outside the app. Maybe a new idea for the JOSM plugin developer?

**Comment by HolgerJeromin** [comment]
I like the idea. Right now I have to make an image and make an osm note with a second app if I want to change something later in the map. 
With this feature I could skip the app switch. 

If you want to fix the problem later by yourself, there could be another work flow :
Clicking somewhere  in the app. Take a photo and get a special reminder email if the image is online.

**Comment by vincentdephily** [comment]
Like @kaitu, I think it should be possible to add a short text directly from the app. Some cases (like a photo of just a street sign) are self-explanatory, so I don't think that adding text should be mandatory.

Of course this could be implemented as a standalone app. Maybe that feature would fit an app like OSMand or OSMTracker better, I'm not sure. But storing both OSM and mapillary credentials , storing the photo long enough to upload, etc seems like a fair amount of work that is mostly covered by the Mapillary app. And doing the last step (creating the osm note) from the Mapillary servers during postprocessing) has a certain appeal to it.

Speaking of creating the note via the Mapillary servers, this implies that Mapillary itself (not the app) is authorised via OAuth to post notes as that OSM user. Not sure which is easyer to implement, but implementing this server-side opens up future possibilities, and is one thing that won't need to be implemented multimple times for each platform :)

**Comment by jesolem** [comment]
Let’s continue discussing features in our forum! We will import this issue as topic to the Mapillary Forum in the coming weeks. We’ll notify here once it’s been done. See you there!
